src/ir/ty.rs

```diff
@@ -356,6 +356,26 @@ impl Type {
             _ => false,
         }
     }
+
+    /// If this type has a known size, return it (in bytes).
+    pub fn calc_size(&self, ctx: &BindgenContext) -> Option<usize> {
+        match self.kind {
+            TypeKind::Comp(ref ci) => ci.calc_layout(ctx).map(|layout| layout.size),
+            TypeKind::Enum(ref enum_ty) => enum_ty.calc_layout(ctx).map(|layout| layout.size),
+            TypeKind::Int(int_kind) => int_kind.known_size(),
+            TypeKind::Float(float_kind) => Some(float_kind.known_size()),
+            TypeKind::Complex(float_kind) => Some(float_kind.known_size() * 2),
+            TypeKind::Reference(..) |
+            TypeKind::NullPtr |
+            TypeKind::Pointer(..) |
+            TypeKind::BlockPointer => Some(mem::size_of::<*mut ()>()),
+            TypeKind::ResolvedTypeRef(inner) |
+            TypeKind::Alias(inner) |
+            TypeKind::TemplateAlias(inner, _) |
+            TypeKind::TemplateRef(inner, _) => ctx.resolve_type(inner).calc_size(ctx),
+            _ => None,
```

Comment:
Please list the other types explicitly here so when we add more people don't forget to add it here.